addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/automation/ReplacerJobUnitTest.java

```diff
@@ -0,0 +1,556 @@
+    }
+
+    @Test
+    void shouldReportInvalidMatchType() {
+        // Given
+        ReplacerJob job =
+                createReplacerJob(
+                        "parameters:\n"
+                                + "  deleteAllRules: false\n"
+                                + "rules:\n"
+                                + "  - matchType: blah\n"
+                                + "    matchString: test");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(
+                progress.getErrors().get(0), is(equalTo("!replacer.automation.error.badmatch!")));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+    }
+
+    @Test
+    void shouldReportBlankMatchString() {
+        // Given
+        ReplacerJob job =
+                createReplacerJob(
+                        "rules:\n" + "  - matchType: resp_body_str\n" + "    matchString: ");
+        AutomationProgress progress = new AutomationProgress();
+
+        // When
+        job.verifyParameters(progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(true)));
+        assertThat(progress.getErrors().size(), is(equalTo(1)));
+        assertThat(
+                progress.getErrors().get(0),
+                is(equalTo("!replacer.automation.error.nomatchstring!")));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+    }
+
+    @Test
+    void shouldRemoveRules() {
+        // Given
+        Model model = mock(Model.class, withSettings().defaultAnswer(CALLS_REAL_METHODS));
+        Model.setSingletonForTesting(model);
+        ExtensionLoader extensionLoader =
+                mock(ExtensionLoader.class, withSettings().strictness(Strictness.LENIENT));
+        ExtensionReplacer extRep =
+                mock(ExtensionReplacer.class, withSettings().strictness(Strictness.LENIENT));
+        given(extensionLoader.getExtension(ExtensionReplacer.class)).willReturn(extRep);
+        ReplacerParam params = mock(ReplacerParam.class);
+        given(extRep.getParams()).willReturn(params);
+        Control.initSingletonForTesting(Model.getSingleton(), extensionLoader);
+
+        ReplacerJob job =
+                createReplacerJob("parameters:\n" + "  deleteAllRules: true\n" + "rules: []");
+        AutomationProgress progress = new AutomationProgress();
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+
+        // When
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        verify(params, times(1)).clearRules();
+    }
+
+    @Test
+    void shouldAddRules() {
```

Comment:
Maybe it should replace the rules with the same name rather than adding it again, the UI/API does not allow duplicates. Not a blocker IMO.
In the dialogue it should probably validate no duplicated descriptions.

Reply:
I kind of think dup descriptions is a user problem - if they cant tell which rule is which because they've used the same description then thats their problem and they should fix it. If its all automated then who cares if the names/descriptions are the same?

Reply:
Sounds good.